include/libopencm3/nrf5/common/temp_common.h

```diff
@@ -0,0 +1,142 @@
+/** @addtogroup uarte_defines TEMP Defines
+ *
+ * @author @htmlonly &copy; @endhtmlonly 2018 
+ * Gerrit Maus <funk@maus.xyz>, Luong Le <novercy@live.com>
+ *
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2018 Gerrit Maus <funk@maus.xyz>, Luong Le <novercy@live.com>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* THIS FILE SHOULD NOT BE INCLUDED DIRECTLY, BUT ONLY VIA TEMP.H
+The order of header inclusion is important. temp.h includes the device
+specific memorymap.h header before including this header file.*/
+
+/** @cond */
+#if defined(LIBOPENCM3_TEMP_H)
+/** @endcond */
+#ifndef LIBOPENCM3_TEMP_COMMON_H
+#define LIBOPENCM3_TEMP_COMMON_H
+
+/**@{*/
+
+/** @defgroup temp_registers TEMP Registers
+ * @ingroup temp_defines
+ * @brief TEMP Registers
+@{*/
+/* --------------------- TEMP registers for NRF52 -------------------------- */
+
+/* Tasks Registers */
+#define TEMP_TASKS_START                 MMIO32((TEMP_BASE) + 0x000)
+#define TEMP_TASKS_STOP                  MMIO32((TEMP_BASE) + 0x004)
+/* Events Registers */
+#define TEMP_EVENTS_DATARDY              MMIO32((TEMP_BASE) + 0x100)
+/* Registers */
+#define TEMP_INTENSET                    MMIO32((TEMP_BASE) + 0x304)
+#define TEMP_INTENCLR                    MMIO32((TEMP_BASE) + 0x308)
+#define TEMP_TEMP                        MMIO32((TEMP_BASE) + 0x508)
+#define TEMP_A                           PTRIO32((TEMP_BASE) + 0x520)
+#define TEMP_A0                          MMIO32((TEMP_BASE) + 0x520)
```

Comment:
this have often just been defined as TEMPA *TEMPA0, is there any real need for the PTRIO thing? or TEMP_A itself?

Reply:
TEMP_A doesn't need PTRIO, just an option to shorten functions later, since offset of Register A1 and A0 has a gap of 4 Bytes, and so on. If libopencm3 uses only Definition, then TEMP_A will be removed.